### PR TITLE
Refactor encoding packages to be nested under common package

### PR DESCRIPTION
**Pull request by jasdel**
Updates the encoding packages, `httpbinding`, `json` and `xml` to all be nested under a common package, `encoding`

### CODE REVIEW

**Comment by skotambkar** [comment]
why is `time` under encoding?

**Comment by jasdel** [comment]
time was moved mainly because it had encoding and decode helpers. :/ though that doesn't fit well with the `SleepWithContext`. Will take a look at this.

File: encoding/doc.go [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// Package encoding provides utilities for encoding values for specific
+// document encodings.
+
+package encoding
```
Comment (skotambkar) [inline encoding/doc.go:4]: Nit: At a later point, we can consider updating this page to have link to available encoding packages

File: mediatype.go [hunk 2]
```diff
@@ -1,9 +0,0 @@
-package smithy
-
-// The MediaType interface is intended to be implemented by string and
-// byte array types whose values can be defined by RFC638 media types.
-type MediaType interface {
-	// Describes the contents of the string or byte array using a media type
-	// as defined by RFC6838.
-	MediaType() string
-}
```
Comment (skotambkar) [inline mediatype.go:9]: Intentional?

**Review by skotambkar — APPROVED** [review]

File: mediatype.go (hunk 2 above)
Comment (jasdel) [reply mediatype.go:9]: yeah this type wasn't used at all and was an artifact that was never deleted from an old design.

**Pull request merged**